requests/utils.py

```diff
@@ -579,6 +581,25 @@ def set_environ(env_name, value):
             os.environ[env_name] = old_value
 
 
+def _proxy_bypass_cached(netloc):
+    """
+    Looks for netloc in the cache,
+    if not found, will call proxy_bypass
+    for the netloc and store its result
+    in the cache
+
+    :rtype: bool
+    """
+
+    cached_result = PROXY_BYPASS_CACHE[netloc]
+
+    if not cached_result:
+        result = proxy_bypass(netloc)
+        PROXY_BYPASS_CACHE[netloc] = result
+        return result
+    else:
+        return cached_result
```

Comment:
My suggestion is that this be reimplemented as a generic decorator, rather than written to specifically handle `proxy_bypass`. That way we get a much easier time of testing the timed cache and all the relevant code. =)